tests/ops/qubit/test_non_parametric_ops.py

```diff
@@ -226,6 +227,22 @@ def test_CY_decomposition(self, tol):
         decomposed_matrix = np.linalg.multi_dot(mats)
         assert np.allclose(decomposed_matrix, op.matrix(), atol=tol, rtol=0)
 
+    def test_CH_decomposition(self, tol):
+        """Tests that the decomposition of the CH gate is correct"""
+        op = qml.CH(wires=[0, 1])
+        res = op.decomposition()
+
+        mats = []
+        for i in reversed(res):
+            if i.wires == Wires([1]):
+                mats.append(np.kron(np.eye(2), i.matrix()))
+            elif i.wires == Wires([0, 1]) and i.name == "CZ":
+                mats.append(np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, -1]]))
```

Comment:
might it be further decomposed than suggested by the class method? I'd make the tests as specific as possible since we should know exactly how it will be decomposed. At a minimum, I'd add an `else: raise Exception("something")` if nothing else should happen

Reply:
The decomposition is `[qml.RY(-np.pi / 4, wires=wires[1]), qml.CZ(wires=wires), qml.RY(+np.pi / 4, wires=wires[1]) ]`
which can probably further decomposed by using decomposition on single elements. Do you mean we need to test that?

Reply:
Good point, I changed it. Thank you!